### PR TITLE
persist schema object during encodeForDB

### PR DESCRIPTION
**Pull request by urbanishimwe**
https://tyktech.atlassian.net/browse/TT-5547

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

### CODE REVIEW

**Comment by jeffy-mathew** [comment]
@urbanishimwe `schema` might be explicitly set to nil after base64 encoding for some reason. I think the better option is to check if `schemaB64` exists and decode it to schema during oas fill method. 
@furkansenharputlu thoughts?

**Comment by urbanishimwe** [comment]
> @urbanishimwe `schema` might be explicitly set to nil after base64 encoding for some reason. I think the better option is to check if `schemaB64` exists and decode it to schema during oas fill method.
> @furkansenharputlu thoughts?

I guess they only wanted to save space, if you peek usages of that method!

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4095/merge
Commit:  
Triggered by: pull_request (@urbanishimwe)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2419791365)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4095/merge
Commit: 0d4270516a91568175d66e92359ce2ddedc38169 
Triggered by: pull_request (@urbanishimwe)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2419800817)

**Review by furkansenharputlu — COMMENTED** [review]
Can you please add a test which reproduces the bug?

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4095/merge
Commit: 710101576ea5fed49165d164c3566fb7b9c4391a 
Triggered by: pull_request (@urbanishimwe)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2420828882)

**Review by titpetric — APPROVED** [review]

File: apidef/api_definition_test.go [hunk 1]
```diff
@@ -56,6 +56,18 @@ func TestEncodeForDB(t *testing.T) {
 		assert.Equal(t, defaultScopeName, spec.JWTScopeClaimName)
 		assert.Equal(t, scopeToPolicyMap, spec.JWTScopeToPolicyMapping)
 	})
+
+	t.Run("EncodeForDB persist schema objects from extended path", func(t *testing.T) {
+		spec := DummyAPI()
+		spec.EncodeForDB()
+		var schemaNotEmpty bool
```
[the remark this answers is not in the document]
Comment (urbanishimwe) [reply apidef/api_definition_test.go:63]: yes, it was the intention, just any of the ones in DummyAPI

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4095/merge
Commit: 1a03874924522acdc09e36cbbb36b016b42fea41 
Triggered by: pull_request (@urbanishimwe)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2426619537)

**Review by buger — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: ee003ba6cdb0aa8f3e7111604e86a1f9e5260a25 persist schema object during encodeForDB (#4095)

* Persist schema object

[changelog]
fixed: schema object to persist during encodeForDB process
Triggered by: push (@lghiur)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2426864167)

**Pull request merged**